catalogue_pipeline/reindexer_v2/reindex_shard_generator/src/shard_manager.py

```diff
@@ -0,0 +1,24 @@
+# -*- encoding: utf-8
+"""
+Contains logic for building reindex shards.
+
+A reindex shard is of the form "{source}/{id}", where "source" is the
+name of the data source (e.g. "miro", "sierra"), and "id" is a shard number string
+based on the ID.
+
+The length of the shard varies based on the data source, to keep individual
+reindex shards for each source at a manageable size.
+
+"""
+
+import math
+
+
+def create_reindex_shard(source_name, source_id, source_size, shard_size):
+    ascii_source_id = int(''.join(str(ord(c)) for c in source_id))
```

Comment:
Neat. I’d been thinking of something slightly different – `sum(ord(c) …)` – but I’m not sure it makes a huge amount of difference.